### PR TITLE
Chat mute player btn, stored locally

### PR DESCRIPTION
**Pull request by reyandme**
Instead of refactoring PR #154 I created a new one by branching previous one at the proper point, it was much easier.

### CODE REVIEW

File: src/net/KM_Networking.pas [hunk 1]
```diff
@@ -1970,8 +1977,12 @@ procedure TKMNetworking.PacketRecieve(aNetClient: TKMNetClient; aSenderIndex: In
                       ChatSound := csChatWhisper;
                       I := NetPlayers.ServerToLocal(PlayerIndex);
                       if I <> -1 then
-                        tmpStringA := NetPlayers[I].Nikname
-                      else
+                      begin
+                        tmpStringA := NetPlayers[I].NiknameColored;
+                        tmpInteger := gResTexts[TX_CHAT_WHISPER_TO].IndexOf('%s');
```
Comment (lewinjh) [inline src/net/KM_Networking.pas]: tmpInteger is not used, unintentional line?

Comment (reyandme) [reply src/net/KM_Networking.pas]: yes. Its always better to have an extra pair of eyes :)

**Comment by lewinjh** [comment]
Looks good to me. Apologies for the magic numbers -1, -2, -3, not sure what I was thinking years ago when I wrote that code, thanks for replacing them with constants hahah.

@Kromster80 you happy with this?

File: src/KM_GameInputProcess.pas [hunk 2]
```diff
@@ -435,6 +426,28 @@ procedure TGameInputProcess.ExecCommand(aCommand: TGameInputCommand);
 end;
 
 
+procedure TGameInputProcess.ExecGameAlertBeaconCmd(aCommand: TGameInputCommand);
+var IsPlayerMuted: Boolean;
+begin
+  with aCommand do
```
Comment (Kromster80) [inline src/KM_GameInputProcess.pas]: Please avoid using `with *** do` constructs in future

File: src/gui/pages_menu/KM_GUIMenuLobby.pas [hunk 3]
```diff
@@ -1291,10 +1367,24 @@ procedure TKMMenuLobby.Lobby_OnPlayersSetup(Sender: TObject);
...
   end;
 
+  // Players flag hightlight, if they are clickable
   for I := 1 to MAX_LOBBY_SLOTS do
     Image_LobbyFlag[I].HighlightOnMouseOver := CanShowPlayerMenu(Image_LobbyFlag[I]);
 
-  //Update the minimap preivew with player colors
+  // Darken player flag when muted
+  for I := 1 to MAX_LOBBY_SLOTS do
+  begin
```
Comment (Kromster80) [inline src/gui/pages_menu/KM_GUIMenuLobby.pas]: No need for begin/end here

File: src/net/KM_Networking.pas [hunk 4]
```diff
@@ -275,6 +280,7 @@ constructor TKMNetworking.Create(const aMasterServerAddress: string; aKickTimeou
   fServerQuery := TKMServerQuery.Create(aMasterServerAddress);
   fNetGameOptions := TKMGameOptions.Create;
   fFileSenderManager := TKMFileSenderManager.Create;
+  fMutedPlayersList := TList.Create;
```
Comment (Kromster80) [inline src/net/KM_Networking.pas:283]: I see construction code. Check if you forgot to add destructor.

File: src/res/KM_Resource.pas [hunk 5]
```diff
@@ -69,6 +69,8 @@   TKMResource = class
     procedure ExportTreeAnim;
     procedure ExportHouseAnim;
     procedure ExportUnitAnim;
+
+    function GetMaxPrintWidthOfStrings(aStrings: array of string; aFont: TKMFont): Integer;
```
Comment (Kromster80) [inline src/res/KM_Resource.pas]: This should be in TKMResFonts, since it uses only Fonts info. Somewhere next to "TKMFontData.GetTextSize"

**Review by Kromster80 — CHANGES_REQUESTED** [review]
I approve these PR, but please see into fixing 3rd and 4th comments

**Comment by reyandme** [comment]
@Kromster80 fixed

**Comment by Kromster80** [comment]
Awesome, thank you!

**Pull request merged**